R/data_unite.R

```diff
@@ -0,0 +1,106 @@
+#' @title Unite ("merge") multiple variables
+#' @name data_unite
+#'
+#' @description
+#' Merges values of multiple variables per observation into one new variable.
+#'
+#' @param data A data frame.
+#' @param new_column The name of the new column, as a string.
+#' @param separator Separator, as string, to use between values.
+#' @param append Logical, if `FALSE` (default), removes original columns that
+#' were united. If `TRUE`, all columns are preserved and the new column is
+#' appended to the data frame.
+#' @param remove_na Logical, if `TRUE`, missing values (`NA`) are not included
+#' in the united values. If `FALSE`, missing values are represented as `"NA"`
+#' in the united values.
+#' @param ... Currently not used.
+#' @inheritParams find_columns
+#'
+#' @inheritSection center Selection of variables - the `select` argument
+#'
+#' @return `data`, with a newly created variable.
+#'
+#' @examples
+#' d <- data.frame(
+#'   x = 1:3,
+#'   y = letters[1:3],
+#'   z = 6:8
+#' )
+#' d
+#' data_unite(d, new_column = "xyz")
+#' data_unite(d, new_column = "xyz", remove = FALSE)
+#' data_unite(d, select = c("x", "z"), new_column = "xyz")
+#' @export
+data_unite <- function(data,
+                       new_column = NULL,
+                       select = NULL,
+                       exclude = NULL,
+                       separator = "_",
+                       append = FALSE,
+                       remove_na = FALSE,
+                       ignore_case = FALSE,
+                       verbose = TRUE,
+                       regex = FALSE,
+                       ...) {
+  # we need a name for the new column
+  if (is.null(new_column)) {
+    insight::format_error(
+      "No name for the new column was provided.",
+      "Please use `new_column` to define a name for the newly created column."
+    )
+  }
+
+  # only one column name
+  if (length(new_column) > 1) {
+    insight::format_error(
+      "Please provide only a single string for `new_column`, no character vector with multiple values."
+    )
+  }
+
+  # evaluate select/exclude, may be select-helpers
+  select <- .select_nse(select,
+    data,
+    exclude,
+    ignore_case,
+    regex = regex,
+    verbose = verbose
+  )
+
+  if (is.null(select) || length(select) <= 1) {
+    insight::format_error(
+      "At least two columns in `select` are required for `data_unite()`."
+    )
+  }
+
+  # unite
+  out[[new_column]] <- do.call(paste, c(d[select], sep = separator))
```

Comment:
my suggestion earlier was incomplete, need to define `out` and replace `d` by `data` here

Reply:
Just realized this when writing tests ;-)